### ISSUE TITLE
Run validation script at the end of `prepare_ml_data.py`?

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/nowcasting_dataset#317]
## Detailed Description
Maybe we should _always_ validated the on-disk batches?

**Comment by peterdudfield** [comment]
The validation script is 'abit' / 'a lot' out of data. Itll need some work to update. The good thing is the Batch validates each data source as we go. 

Perhaps a easy cahnge to make, would be to validate the t0_datetimes are in sperate groups for the train, validation and test

**Comment by JackKelly** [comment]
> validate the t0_datetimes are in sperate groups for the train, validation and test

I completely agree!  I think I implemented this here: https://github.com/openclimatefix/nowcasting_dataset/blob/main/nowcasting_dataset/dataset/split/split.py#L189
